firewood/src/merkle/node/leaf.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use std::fmt::{Debug, Error as FmtError, Formatter};
+
+use bincode::Options;
+
+use super::{Data, Encoded};
+use crate::merkle::{from_nibbles, PartialPath};
+
+#[derive(PartialEq, Eq, Clone)]
+pub struct LeafNode {
+    pub(crate) path: PartialPath,
+    pub(crate) data: Data,
+}
+
+impl Debug for LeafNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), FmtError> {
+        write!(f, "[Leaf {:?} {}]", self.path, hex::encode(&*self.data))
+    }
+}
+
+impl LeafNode {
+    pub fn new<P: Into<PartialPath>, D: Into<Data>>(path: P, data: D) -> Self {
+        Self {
+            path: path.into(),
+            data: data.into(),
+        }
+    }
+
+    pub fn path(&self) -> &PartialPath {
+        &self.path
+    }
+
+    pub fn data(&self) -> &Data {
+        &self.data
+    }
+
+    pub(super) fn encode(&self) -> Vec<u8> {
+        bincode::DefaultOptions::new()
+            .serialize(
+                [
+                    Encoded::Raw(from_nibbles(&self.path.encode(true)).collect()),
+                    Encoded::Raw(self.data.to_vec()),
+                ]
+                .as_slice(),
+            )
+            .unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_case::test_case;
+
+    // these tests will fail if the encoding mechanism changes and should be updated accordingly
+    #[test_case(0b10 << 4, vec![0x12, 0x34], vec![1, 2, 3, 4]; "even length")]
+    // first nibble is part of the prefix
+    #[test_case((0b11 << 4) + 2, vec![0x34], vec![2, 3, 4]; "odd length")]
+    fn encode_regression_test(prefix: u8, path: Vec<u8>, nibbles: Vec<u8>) {
```

Comment:
nit: at a minimum add a comment in the PR  that this is a new, not moved, test

Reply:
Is this just testing `partial_path`? If you're checking to see if the node is storing it correctly, rather than hardcoding this here (which can change if partial_path changes), perhaps use partial_path and compare that encoding?

Reply:
Yes, it looks like this is just testing partial-path

I've created https://github.com/ava-labs/firewood/issues/365 to move the test